### PR TITLE
Adds pagination configuration example

### PR DESCRIPTION
**Pull request by bravo-kernel**
- adds pagination listener configuration example (for new users)
- fixes wrong class in pagination `AppController` example
- adds missing event for all `beforeFilter()` examples (all docs)

### CODE REVIEW

File: docs/listeners/redirect.rst [hunk 1]
```diff
@@ -84,7 +84,7 @@ The closure takes two arguments:
   <?php
   class SamplesController extends AppController {
 
-    public function beforeFilter() {
+    public function beforeFilter(\Cake\Event\Event $event) {
```
Comment (josegonzalez) [inline docs/listeners/redirect.rst]: Move this to a use statement please :)

File: docs/listeners/related-models.rst [hunk 2]
```diff
@@ -113,7 +113,7 @@ Example
 	<?php
 	class DemoController extends AppController {
 
-		public function beforeFilter() {
+		public function beforeFilter(\Cake\Event\Event $event) {
```
Comment (josegonzalez) [inline docs/listeners/related-models.rst]: Here too!

File: docs/listeners/api-pagination.rst [hunk 3]
```diff
@@ -10,45 +10,46 @@ pagination information.
...
 
-		public $components = [
-			'RequestHandler',
-			'Crud.Crud' => [
-				'listeners' => [
-					'Crud.Api', // Required
-					'Crud.ApiPagination'
-				]
-			];
+		public function beforeFilter(\Cake\Event\Event $event) {
```
Comment (josegonzalez) [inline docs/listeners/api-pagination.rst]: samesies

File: docs/listeners/api-pagination.rst [hunk 4]
```diff
@@ -10,45 +10,46 @@ pagination information.
...
-Attach it on the fly in your controller beforeFilter, this is recommended if
-you want to attach it only to specific controllers and actions
+Attach this listener to your AppController components array if you want to make
+it available for all your controllers, application wide.
 
 .. code-block:: php
 
 	<?php
-	class SamplesController extends AppController {
+	class AppController extends \Cake\Controller\Controller {
```
Comment (josegonzalez) [inline docs/listeners/api-pagination.rst]: again here.

**Comment by bravo-kernel** [comment]
I don't think so Mr. @josegonzalez, these changes make the examples consistent with all other examples throughout the docs.

**Comment by josegonzalez** [comment]
:(

**Comment by bravo-kernel** [comment]
I do see your point and it has made me wonder before but I believe it should be a separate PR if "you friends" feel strongly enough about it.

**Pull request merged**